src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,545 @@
+            }
+            else {
+                ret.push(getSynthesizedDeepClone(stmt));
+            }
+        }
+
+        return createNodeArray(ret);
+    }
+
+
+    function getInnerTransformationBody(transformer: Transformer, innerRetStmts: Node[], prevArgName?: SynthIdentifier) {
+
+        let innerCbBody: Statement[] = [];
+        for (const stmt of innerRetStmts) {
+            forEachChild(stmt, function visit(node: Node) {
+                if (isCallExpression(node)) {
+                    const temp = transformExpression(node, transformer, node, prevArgName);
+                    innerCbBody = innerCbBody.concat(temp);
+                    if (innerCbBody.length > 0) {
+                        return;
+                    }
+                }
+                else if (!isFunctionLike(node)) {
+                    forEachChild(node, visit);
+                }
+            });
+        }
+        return innerCbBody;
+    }
+
+    function hasPropertyAccessExpressionWithName(node: CallExpression, funcName: string): boolean {
+        if (!isPropertyAccessExpression(node.expression)) {
+            return false;
+        }
+
+        return node.expression.name.text === funcName;
+    }
+
+    function getArgName(funcNode: Node, transformer: Transformer): SynthIdentifier {
+
+        const numberOfAssignmentsOriginal = 0;
+        const types: Type[] = [];
+
+        let name: SynthIdentifier | undefined;
+
+        if (isFunctionLikeDeclaration(funcNode)) {
+            if (funcNode.parameters.length > 0) {
+                const param = funcNode.parameters[0].name as Identifier;
+                name = getMapEntryIfExists(param);
+            }
+        }
+        else if (isCallExpression(funcNode) && funcNode.arguments.length > 0 && isIdentifier(funcNode.arguments[0])) {
+            name = { identifier: funcNode.arguments[0] as Identifier, types, numberOfAssignmentsOriginal };
+        }
+        else if (isIdentifier(funcNode)) {
+            name = getMapEntryIfExists(funcNode);
+        }
+
+        if (!name || name.identifier === undefined || name.identifier.text === "_" || name.identifier.text === "undefined") {
+            return { identifier: createIdentifier(""), types, numberOfAssignmentsOriginal };
+        }
+
+        return name;
+
+        function getMapEntryIfExists(node: Identifier): SynthIdentifier {
+            const originalNode = getOriginalNode(node);
+            const symbol = getSymbol(originalNode);
+            const identifier = node;
+
+            if (!symbol) {
+                return { identifier, types, numberOfAssignmentsOriginal };
+            }
+
+            const mapEntry = transformer.synthNamesMap.get(getSymbolId(symbol).toString());
+            return mapEntry || { identifier, types, numberOfAssignmentsOriginal };
+        }
+
+        function getSymbol(node: Node): Symbol | undefined {
+            return node.symbol ? node.symbol : transformer.checker.getSymbolAtLocation(node);

```

Comment:
Why not check `node.symbol` in the other places where `getSymbolAtLocation` is called (or in `getSymbolAtLocation` itself)?  Note: this is a question, not a prompt - I don't understand the difference between the two mechanisms.